Core/Events/PathOfTerrariaPlayer.Events.cs

```diff
@@ -58,19 +58,33 @@ public override void OnHitByProjectile(Projectile proj, Player.HurtInfo hurtInfo
 	{
 		OnHitByProjectileEvent?.Invoke(Player, proj, hurtInfo);
 	}
+	
+	/// <summary>
+	/// Use this event for the Player hitting an NPC
+	/// This happens before the onHit hook and should be used if the effect modifies the any of the ref variables otherwise stick to the onHit.
+	/// call StarlightPlayer.SetHitPacketStatus to sync if this has an effect beyond editting ref variables.
```

Comment:
StarlightPlayer does not exist. This system is ported from SLR, so that's why it's here, but this would be very confusing to read as documentation.
This happens a lot of times too.

Reply:
I removed all references from StarlightPlayer. This was copied in from a long while ago when it was just Scalar and myself.